ci/util/serial_interface.py

```
@@ -140,14 +141,25 @@ async def read_lines(self, timeout: float) -> AsyncIterator[str]:
+                while True:
+                    remaining = deadline - time.monotonic()
+                    if remaining <= 0:
+                        break
+                    for line in self._monitor.read_lines(timeout=remaining):
+                        loop.call_soon_threadsafe(queue.put_nowait, line)
             except KeyboardInterrupt:
                 _thread.interrupt_main()
                 raise
-            except Exception:
-                pass
+            except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)